### PR TITLE
feat(#973): Allow standard reporter override

### PR DESCRIPTION
**Pull request by tschlat**
I would like to provide a solution for the respective issue.

Note: I am not sure about the comment, I added to the docs as I am not familliar with the configuration of Citrus context in the case that Spring is not used.

### CODE REVIEW

File: core/citrus-base/src/main/java/org/citrusframework/report/DefaultTestReporters.java [hunk 1]
```diff
@@ -1,21 +1,18 @@
...
 public class DefaultTestReporters extends TestReporters {
 
-    public static final List<TestReporter> DEFAULT_REPORTERS = Arrays.asList(
-            new LoggingReporter(),
-            new HtmlReporter(),
-            new JUnitReporter()
-    );
+    public static final Map<String, TestReporter> DEFAULT_REPORTERS = Map.of("citrusLoggingReporter", new LoggingReporter(),
+        "citrusHtmlReporter",new HtmlReporter(),
+        "citrusJunitReporter",new JUnitReporter());
```
Comment (bbortt) [inline core/citrus-base/src/main/java/org/citrusframework/report/DefaultTestReporters.java:13]:
```suggestion
    public static final Map<String, TestReporter> DEFAULT_REPORTERS = Map.of(
        "citrusLoggingReporter", new LoggingReporter(),
        "citrusHtmlReporter", new HtmlReporter(),
        "citrusJunitReporter", new JUnitReporter()
    );
```

**Comment by bbortt** [comment]
I think you'll also need to fix some tests: https://github.com/citrusframework/citrus/actions/runs/6106192495/job/16570862788?pr=974#step:6:1751.

**Comment by tschlat** [comment]
> I think you'll also need to fix some tests: https://github.com/citrusframework/citrus/actions/runs/6106192495/job/16570862788?pr=974#step:6:1751.

**Comment by tschlat** [comment]
> > I think you'll also need to fix some tests: https://github.com/citrusframework/citrus/actions/runs/6106192495/job/16570862788?pr=974#step:6:1751.

I saw this comming. Now the beans are missing and cannot be autowired. Unforunately, these Reporters also implement some other interfaces like TestListener, TestSuiteListener, TestActionListener and MessageListener.  
So I see, that this approach is suboptimal.

**Comment by bbortt** [comment]
closes https://github.com/citrusframework/citrus/issues/973.

**Comment by tschlat** [comment]
Closed in favour of: [973](https://github.com/citrusframework/citrus/pull/973)

**Pull request closed without merging**